packages/@sanity/base/src/components/validation/validationList.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useCallback} from 'react'
+import {
+  ObjectSchemaType,
+  Path,
+  Marker,
+  isValidationErrorMarker,
+  isValidationWarningMarker,
+} from '@sanity/types'
+import {Box} from '@sanity/ui'
+import {ListItem} from './listItem'
+
+interface ValidationListProps {
+  documentType?: ObjectSchemaType
+  kind?: 'simple'
+  markers: Marker[]
+  onFocus?: (path: Path) => void
+  onClose?: () => void
+  // showLink?: boolean
+  truncate?: boolean
+}
+
+export function ValidationList(props: ValidationListProps) {
+  const {
+    documentType,
+    kind,
+    markers,
+    onFocus,
+    onClose,
+    truncate,
+    // showLink,
+  } = props
+
+  const errors = markers.filter(isValidationErrorMarker)
+  const warnings = markers.filter(isValidationWarningMarker)
+
+  const handleClick = useCallback(
+    (path: Path = []) => {
+      if (onFocus) onFocus(path)
+      if (onClose) onClose()
+    },
+    [onFocus, onClose]
+  )
+
+  const resolvePathTitle = (path: Path) => {
+    const fields = documentType && documentType.fields
+    const field = fields && fields.find((curr) => curr.name === path[0])
+
+    return (field && field.type.title) || ''
+  }
+
+  const hasErrors = errors.length > 0
+  const hasWarnings = warnings.length > 0
+
+  if (!hasErrors && !hasWarnings) {
+    return null
+  }
+
+  return (
+    <Box data-kind={kind} style={{maxWidth: 250, maxHeight: '60vh', overflow: 'auto'}}>
```

Comment:
Can we wrap this into a component, or lift the styles to a static object to simplify re-rendering checks?

Reply:
Yes, will do 👍